packages/ra-keycloak/src/authProvider.ts

```diff
@@ -111,7 +111,7 @@ export const keycloakAuthProvider = (
         });
     },
     async logout() {
-        let redirectUri = window.location.origin;
+        let redirectUri = `${window.location.origin}/login`;
```

Comment:
This is actually what fixed the redirect loop: the Keycloak client was constantly redirecting to http://localhost:8081/, and hence the LoginPage useEffect didn't have time to fire most of the time.